### DISCUSSION TITLE
<Head> in _document.js or _app.js

### DISCUSSION THREAD

**Discussion by bugproof** [vercel/next.js#24452] [Help]
If I can put all `<head>` related tags in `_app.js` what benefits do I get by putting them in `_document.js` instead?

`_document.js` has a limit and you can't add viewport metatag and title tag there and `_app.js` doesn't have any limits regarding this

I also asked this question on SO but didn't get any logical answer https://stackoverflow.com/questions/67242285/head-in-document-js-or-app-js/67242785#67242785

**Comment by TheDarkStrix** [comment]
![image](https://user-images.githubusercontent.com/18639879/116062008-2378d900-a6a1-11eb-9aff-9b47d08defcb.png)

The only time I would use _document.js is to edit the default html markup , only _document.js lets you override the very top-level document of your application, which renders the <html> element , which is same on all the page of the projects.

Eg: 
```
in _document.js 
<html lang="en">
...
<body className="bodyClass">
</body>
</html>
```

Please note : _document only runs on the server and not on the client side.

Considering that usually content inside the head tag change from page to page (eg; title) , its is better to include them within pages (this runs on the client side).
